addons/mail/models/models.py

```diff
@@ -20,13 +20,45 @@ class Base(models.AbstractModel):
     _inherit = 'base'
     _mail_defaults_to_email = False
 
+    # ------------------------------------------------------------
+    # ORM
+    # ------------------------------------------------------------
+
     def _valid_field_parameter(self, field, name):
         # allow tracking on abstract models; see also 'mail.thread'
         return (
             name == 'tracking' and self._abstract
             or super()._valid_field_parameter(field, name)
         )
 
+    # ------------------------------------------------------------
+    # CHECK ACCESS
+    # ------------------------------------------------------------
+
+    def _group_mail_message_access(self, message_access):
+        """ mail.message check permission rules based on related document. This
```

Comment:
To myself: confusing. It shoudl be "document permission check based on their mail.message permission check".